src/textures/Texture.js

```diff
@@ -126,7 +126,9 @@ Object.assign( Texture.prototype, EventDispatcher.prototype, {
 
 	toJSON: function ( meta ) {
 
-		if ( meta.textures[ this.uuid ] !== undefined ) {
+		var isRootObject = (meta === undefined || typeof meta === 'string' );
+
+		if ( !isRootObject && meta.textures[ this.uuid ] !== undefined ) {
```

Comment:
Unary operator '!' must be followed by whitespace. These happens several times in your code.

```javascript
if ( ! isRootObject && meta.textures[ this.uuid ] !== undefined ) {
```